kubectl-fdb/cmd/exec.go

```diff
@@ -0,0 +1,165 @@
+ * limitations under the License.
+ */
+
+package cmd
+
+import (
+	ctx "context"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+
+	"github.com/spf13/cobra"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/selection"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	fdbtypes "github.com/FoundationDB/fdb-kubernetes-operator/api/v1beta1"
+	controllers "github.com/FoundationDB/fdb-kubernetes-operator/controllers"
+)
+
+func newExecCmd(streams genericclioptions.IOStreams) *cobra.Command {
+	o := NewFDBOptions(streams)
+
+	cmd := &cobra.Command{
+		Use:   "exec",
+		Short: "Runs a command on a container in an FDB cluster",
+		Long:  "Runs a command on a container in an FDB cluster",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cluster, err := cmd.Flags().GetString("fdb-cluster")
+			if err != nil {
+				return err
+			}
+
+			config, err := o.configFlags.ToRESTConfig()
+			if err != nil {
+				return err
+			}
+
+			scheme := runtime.NewScheme()
+			_ = clientgoscheme.AddToScheme(scheme)
+			_ = fdbtypes.AddToScheme(scheme)
+
+			kubeClient, err := client.New(config, client.Options{Scheme: scheme})
+			if err != nil {
+				return err
+			}
+
+			namespace, err := getNamespace(*o.configFlags.Namespace)
+			if err != nil {
+				return err
+			}
+
+			context, err := cmd.Flags().GetString("context")
+			if err != nil {
+				return err
+			}
+
+			err = runExec(kubeClient, cluster, context, namespace, args)
+			if err != nil {
+				return err
+			}
+
+			return nil
+		},
+		Example: `
+ # Open a shell.
+ kubectl fdb exec -c cluster
+
+ # Run a status command.
+ kubectl fdb exec -c cluster -- fdbcli --exec "status minimal"
+ `,
+	}
+
+	cmd.Flags().StringP("fdb-cluster", "c", "", "exec into the provided cluster.")
+	err := cmd.MarkFlagRequired("fdb-cluster")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	o.configFlags.AddFlags(cmd.Flags())
+
+	return cmd
+}
+
+func buildCommand(kubeClient client.Client, clusterName string, context string, namespace string, commandArgs []string) (exec.Cmd, error) {
+	pods := &corev1.PodList{}
+
+	clusterRequirement, err := labels.NewRequirement(controllers.FDBClusterLabel, selection.Equals, []string{clusterName})
+	if err != nil {
+		return exec.Cmd{}, nil
+	}
+	processClassRequirement, err := labels.NewRequirement(controllers.FDBProcessClassLabel, selection.Exists, nil)
+	if err != nil {
+		return exec.Cmd{}, nil
+	}
+
+	err = kubeClient.List(ctx.Background(), pods,
+		client.InNamespace(namespace),
+		client.MatchingLabelsSelector{Selector: labels.NewSelector().Add(*clusterRequirement, *processClassRequirement)},
+	)
+	if err != nil {
+		return exec.Cmd{}, err
+	}
+	if len(pods.Items) == 0 {
+		return exec.Cmd{}, fmt.Errorf("No pods found for cluster %s", clusterName)
+	}
+	kubectlPath, err := exec.LookPath("kubectl")
```

Comment:
That's a good idea to use `kubectl` the last time I wanted to use the `exec` "API it was not very well documented and a little bit "experimental". For me using `kubectl` is fine since this is a requirement for the plugin anyway.